Add release notes 2.14.5 (#11196)
PrefectHQ/prefect: RELEASE-NOTES.md

# Prefect Release Notes

## Release 2.14.5

### Highlight TBD

### Enhancements
- Add support for storage blocks to `flow.from_source` — https://github.com/PrefectHQ/prefect/pull/11092
- add settings for worker webserver host and port — https://github.com/PrefectHQ/prefect/pull/11175
- Safely retrieve `flow_run_id` in `EventsWorker` — https://github.com/PrefectHQ/prefect/pull/11182
- Add client-side setting for default work pool — https://github.com/PrefectHQ/prefect/pull/11137
- Add task run tag concurrency slot wait setting — https://github.com/PrefectHQ/prefect/pull/11020

### Fixes
- Fix errors when using `GitRepository` with a private repo — https://github.com/PrefectHQ/prefect/pull/11156
- Fix bug where check for required packages fails incorrectly — https://github.com/PrefectHQ/prefect/pull/11111

> - Fix routing to the flow page from a flow run — https://github.com/PrefectHQ/prefect/pull/11190

Reviewer: ```suggestion
- Fix routing to the flow page from a flow run — https://github.com/PrefectHQ/prefect/pull/11190
- Fix issue where flow runs page wasn't reacting to filter changes - https://github.com/PrefectHQ/prefect-ui-library/pull/1874
```

from prefect_aws import AwsCredentials
from prefect_aws.s3 import S3Bucket

if __name__ == "__main__":
    flow.from_source(
        source=S3Bucket(
            bucket_name="my-code-storage-bucket",
            credentials=AwsCredentials(
                aws_access_key_id="my-access-key-id",
                aws_secret_access_key="my-secret-access-key",
            ),
        ),
        entrypoint="flows.py:my_flow",
    ).deploy(name="my-deployment", work_pool_name="above-ground")
```

Here's an example of loading and deploying a flow from an S3 bucket:

```python
from prefect import flow
from prefect_aws.s3 import S3Bucket

if __name__ == "__main__":
    flow.from_source(
        source=S3Bucket.load("my-code-storage-bucket"), entrypoint="flows.py:my_flow"
    ).deploy(name="my-deployment", work_pool_name="above-ground")
```

Note that a storage block must be saved before deploying a flow, but not if you're serving a remotely stored flow.

See the following pull request for implementation details:
- https://github.com/PrefectHQ/prefect/pull/11092

### Enhancements
- Add customizable host and port settings for worker webserver — https://github.com/PrefectHQ/prefect/pull/11175

> - Safely retrieve `flow_run_id` in `EventsWorker` for finding related events — https://github.com/PrefectHQ/prefect/pull/11182

Reviewer 2: ```suggestion
- Safely retrieve `flow_run_id` in `EventsWorker` while finding related events — https://github.com/PrefectHQ/prefect/pull/11182
```

- Enable enhanced cancellation by default - https://github.com/PrefectHQ/prefect/pull/11192

### Fixes
- Fix access token retrieval when using `GitRepository` with a private repo and `.deploy` — https://github.com/PrefectHQ/prefect/pull/11156
- Fix bug where check for required packages fails incorrectly during `prefect deploy` — https://github.com/PrefectHQ/prefect/pull/11111
- Fix routing to the Flows page from a flow run in the Prefect UI — https://github.com/PrefectHQ/prefect/pull/11190
- Ensure Flow Runs page reacts to filter changes - https://github.com/PrefectHQ/prefect-ui-library/pull/1874
- Optimize memory usage by clearing `args/kwargs` in a Prefect `Call` post-execution -  https://github.com/PrefectHQ/prefect/pull/11153
- Allow logs to handle un-`uuid`-like flow_run_ids - https://github.com/PrefectHQ/prefect/pull/11191
- Only run unit tests for Python file changes — https://github.com/PrefectHQ/prefect/pull/11159
- Add `codespell` config and add to pre-commit  — https://github.com/PrefectHQ/prefect/pull/10893
- Update token regex in release notes generation script for VSCode compatibility - https://github.com/PrefectHQ/prefect/pull/11195

### Documentation

> - Add Terraform Provider to guides, tidy guides navigation — https://github.com/PrefectHQ/prefect/pull/11170

Reviewer 3: ```suggestion
- Add Terraform Provider guide, update and simplify guides navigation — https://github.com/PrefectHQ/prefect/pull/11170
```

> - Update Prefect Cloud documentation to reflect product changes — https://github.com/PrefectHQ/prefect/pull/11157

Reviewer 3: ```suggestion
- Clarify and harmonize Prefect Cloud documentation to reflect nomenclature and UX changes — https://github.com/PrefectHQ/prefect/pull/11157
```
